### ISSUE TITLE
Deleted zone

### ISSUE THREAD

**Issue by BlixLT** [mentrena/SyncKit#67]
Not sure if I am doing something wrong or SyncKit cannot handle deleted record zone (at least in some cases). There are several zones in my cloudkit container (not only the ones that are being used by syncKit), so after deleting the zone that was used by SyncKit, I cannot successfully make synchronization anymore. Should I make some specific actions in this case?

As far as I can see - there is `setupRecordZoneIfNeeded` method, but its not being called if `changedZoneIDs.count > 0` in the `fetchDatabaseChangesWithCompletion` method. I think another check for the empty array after getting toFetchZoneIDs could fix the problem:

```
            if (changedZoneIDs.count) {
...                
                if (toFetchZoneIDs.count)
                {
...
                }
                else
                {
                    callBlockIfNotNil(completion, databaseToken, nil);
                }
            } else {
                callBlockIfNotNil(completion, databaseToken, nil);
            }
```

Without it the sync process will not go to the `setupRecordZoneIfNeeded` method, but just ends in the fetch changes step with the cloudKit error, something like no zoneIDs given to fetchChangesOperation. 

Or maybe there is something else that I should do in such case?

**Comment by mentrena** [comment]
Hi @BlixLT ,

Are you deleting the record zone on one device and then trying to sync on another? Or is this on the same device?
Are you deleting the record zone using SyncKit or using some other method?

**Comment by BlixLT** [comment]
I am not 100% sure as it was a while ago, but I think I deleted that zone using cloudkit dashboard and I also deleted local SyncKit's sqlite files for that zone. And SyncKit stopped working then with the error smth like no zoneIDs given to fetchChangesOperation. The code written above fixed that issue for me. Because changedZoneIDs contained a zone, that is not used by SyncKit (it is used for syncing some other data).
